File: find-lucky-integer-in-an-array/find-lucky-integer-in-an-array.py
class Solution:
    def findLucky(self, arr: List[int]) -> int:
        max_fre = -1
        for i in arr:
            s = arr.count(i)
            if i == s:
                max_num = i 
                if max_fre<max_num:
                    max_fre= max_num
        return max_fre

File: find-lucky-integer-in-an-array/find-lucky-integer-in-an-array.cpp
class Solution {
public:
    int findLucky(vector<int>& arr) {
        unordered_map<int,int>res ;
        for(int i = 0 ; i < arr.size() ; i++){
            res[arr[i]]++;
        }
        int ans = -1 ;
        for(auto i : res){
            if(i.first==i.second){
                ans = max(ans,i.first);
            }
        }
        return ans;
    }
};